update molecular subtyping pathology

jharenza opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
Update molecular subtyping pathology module. NOTE: #851, #853, and #852  should be be merged and run first.

#### What was your approach?
- Update to v17 release
- Add Cranio, EPN, MB, neurocytoma subtyping
- Added integrated_diagnosis, broad_histology, and short_histology so that we don't destroy CI by leaving these all blank before #748 is done. I did this based on [this work in progress document](https://docs.google.com/spreadsheets/d/1fDXt_YODcSAWDvyI5ISBVhUCu4b5-TFCVWMOwiPeMwk/edit#gid=1210552782).

#### What GitHub issue does your pull request address?
#719, #784, #667 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?
- I left `integrated_diagnosis` as NA if there is no subtype
- `PT_AQWDQW27` seems to be missing from the EPN subtyping module and was noted as a sample which is a rare meningioma but was subtyped as EPN. Will create a ticket to assess the EPN module.

#### Is there anything that you want to discuss further?
Note: `PT_KTRJ8TFY` still has two different subtypes for samples which are `Progressive Disease Post-Mortem`- @kgaonkar6 will you check that sample to be sure it should end up this way from clinical feedback? This was one sample which did not have a clinical sequencing report, so this may be expected.

Note: `PT_6Q0NPVP3` also has duplicate subtypes, but this will be taken care of once #851 is merged.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
yes


### Results

#### What types of results are included (e.g., table, figure)?
tables

#### What is your summary of the results?
1497 rows in final table, but this will change with modules being merged ahead



#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

kgaonkar6 commented:
Oh yes, BS_EE73VE7V ( 1 of the sample Progressive Disease Post-Mortem for PT_KTRJ8TFY ) only had the histone variant in Vardict so is lost in consensus calling so got subtyped as H3 wildtype.

Ticket https://github.com/d3b-center/bixu-tracker/issues/463 shows the IGV for the variant, we should probably add that to the clinical/pathology module as H3 mutatant along with other updates?

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -140,23 +158,42 @@ embryonal_results_df <- bind_exp_strategies(embryonal_results_df) %>%
                     broad_histology,
                     Notes),
-             by = "Kids_First_Biospecimen_ID")
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = 
+           ifelse(molecular_subtype == "CNS Embryonal, NOS", "CNS Embryonal Tumor, NOS",
+                  ifelse(molecular_subtype == "CNS HGNET-MN1", "CNS Embryonal Tumor, HGNET-MN1",
+                         ifelse(molecular_subtype == "CNS NB-FOXR2", "CNS neuroblastoma",
+                                ifelse(molecular_subtype == "ETMR, C19MC-altered", "Embryonal tumor with multilayer rosettes, C19MC-altered",
+                                       ifelse(molecular_subtype == "ETMR, NOS",
+                "Embryonal tumor with multilayer rosettes, NOS", NA))))),
+        short_histology = 
+          ifelse(molecular_subtype %in% c("ETMR, C19MC-altered", "ETMR, NOS"),
+                "ETMR", "Embryonal Tumor"),
+        broad_histology = "Embryonal Tumor")
```
```suggestion
             by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "CNS Embryonal, NOS"~ "CNS Embryonal Tumor, NOS",
                                          molecular_subtype == "CNS HGNET-MN1"~ "CNS Embryonal Tumor, HGNET-MN1",
                                          molecular_subtype == "CNS NB-FOXR2" ~ "CNS neuroblastoma",
                                          molecular_subtype == "ETMR, C19MC-altered"~ "Embryonal tumor with multilayer rosettes, C19MC-altered",
                                          molecular_subtype == "ETMR, NOS"~"Embryonal tumor with multilayer rosettes, NOS",
                                          TRUE ~ NA_character_), 
        short_histology = 
          if_else(molecular_subtype %in% c("ETMR, C19MC-altered", "ETMR, NOS"),
                "ETMR", "Embryonal Tumor"),
        broad_histology = "Embryonal Tumor")

```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
                     broad_histology,
                     Notes),
-             by = "Kids_First_Biospecimen_ID") 
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = 
+           ifelse(molecular_subtype == "DMG, H3 K28", "Diffuse midline glioma, H3 K28-mutant",
+         ifelse(molecular_subtype == "DMG, H3 K28, BRAF V600E", "Diffuse midline glioma, H3 K28-mutant, BRAF V600E",
+                ifelse(molecular_subtype == "HGG, BRAF V600E", "High-grade glioma/astrocytoma, BRAF V600E", 
+                       ifelse(molecular_subtype == "HGG, H3 G35", "High-grade glioma/astrocytoma, H3 G35-mutant",
+                         ifelse(molecular_subtype == "HGG, H3 wildtype", "High-grade glioma/astrocytoma, H3 wildtype",
+                                ifelse(molecular_subtype == "HGG, IDH", "High-grade glioma/astrocytoma, IDH-mutant", NA)))))),
+                  broad_histology == "Diffuse astrocytic and oligodendroglial tumor",
+                  short_histology == "HGAT")
```
```suggestion
             by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "DMG, H3 K28"~ "Diffuse midline glioma, H3 K28-mutant",
                                          molecular_subtype == "DMG, H3 K28, BRAF V600E"~ "Diffuse midline glioma, H3 K28-mutant, BRAF V600E",
                                          molecular_subtype == "HGG, BRAF V600E"~ "High-grade glioma/astrocytoma, BRAF V600E",
                                          molecular_subtype == "HGG, H3 G35" ~ "High-grade glioma/astrocytoma, H3 G35-mutant",
                                          molecular_subtype == "HGG, H3 wildtype" ~ "High-grade glioma/astrocytoma, H3 wildtype",
                                          molecular_subtype == "HGG, IDH"~ "High-grade glioma/astrocytoma, IDH-mutant", 
                                          TRUE~ NA_character_),
                  broad_histology = "Diffuse astrocytic and oligodendroglial tumor",
                  short_histology = "HGAT")
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
                     broad_histology,
                     Notes),
-             by = "Kids_First_Biospecimen_ID") 
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = 
+           ifelse(molecular_subtype == "LGG, BRAF fusion", "Low-grade glioma/astrocytoma, BRAF fusion",
+                  ifelse(molecular_subtype == "LGG, BRAF V600E", "Low-grade glioma/astrocytoma, BRAF V600E", 
+                         ifelse(molecular_subtype == "LGG, BRAF wildtype", "Low-grade glioma/astrocytoma, BRAF wildtype", NA))),
+                  broad_histology == "Low-grade astrocytic tumor",
+                  short_histology == ifelse(pathology_diagnosis == "Ganglioglioma", "Ganglioglioma", "LGAT"))
```
```suggestion
             by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "LGG, BRAF fusion"~ "Low-grade glioma/astrocytoma, BRAF fusion",
                                          molecular_subtype == "LGG, BRAF V600E" ~ "Low-grade glioma/astrocytoma, BRAF V600E", 
                                          molecular_subtype == "LGG, BRAF wildtype"~ "Low-grade glioma/astrocytoma, BRAF wildtype", 
                                          TRUE ~NA_character_),
                  broad_histology = "Low-grade astrocytic tumor",
                  short_histology = if_else(pathology_diagnosis == "Ganglioglioma", "Ganglioglioma", "LGAT"))
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
````diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
+                    short_histology,
+                    broad_histology,
+                    Notes),
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(molecular_subtype = subgroup) %>%
+  mutate(molecular_subtype = ifelse(is.na(molecular_subtype), "EPN, To be classified", molecular_subtype), 
+         integrated_diagnosis = ifelse(molecular_subtype == "PT_EPN_A", 
+                                       "Posterior Fossa Ependymoma, Type A",
+                                       ifelse(molecular_subtype == "ST_EPN_RELA",
+                                       "Supratentorial Ependymoma, RELA fusion positive",
+                                       ifelse(molecular_subtype == "ST_EPN_YAP1", "Supratentorial Ependymoma, YAP1 fusion positive", NA))),
+         broad_histology == "Ependymal Tumor",
+         short_histology == "Ependymoma")
 ```
````
```suggestion
           by = "Kids_First_Biospecimen_ID") %>%
  mutate(molecular_subtype = subgroup) %>%
  mutate(molecular_subtype = if_else(is.na(molecular_subtype), "EPN, To be classified", molecular_subtype), 
         integrated_diagnosis = case_when(molecular_subtype == "PT_EPN_A" ~ "Posterior Fossa Ependymoma, Type A",
                                          molecular_subtype == "ST_EPN_RELA" ~ "Supratentorial Ependymoma, RELA fusion positive",
                                          molecular_subtype == "ST_EPN_YAP1" ~ "Supratentorial Ependymoma, YAP1 fusion positive", 
                                          TRUE~ NA_character_),
         broad_histology = "Ependymal Tumor",
         short_histology = "Ependymoma")
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
+                    short_histology,
+                    broad_histology,
+                    Notes),
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = ifelse(molecular_subtype == "MB, SHH",
+                                       "Medulloblastoma, SHH-activated",
+                                       ifelse(molecular_subtype == "MB, WNT","Medulloblastoma, WNT-activated",
+                                              ifelse(molecular_subtype == "MB, Group3", 
+                                                     "Medulloblastoma, group 3",
+                                                     ifelse(molecular_subtype == "MB, Group4",
+                                                            "Medulloblastoma, group 4", NA)))),
+         broad_histology == "Embryonal Tumor",
+         short_histology == "Medulloblastoma")
```
```suggestion
            by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "MB, SHH"~"Medulloblastoma, SHH-activated",
                                          molecular_subtype == "MB, WNT"~"Medulloblastoma, WNT-activated",
                                          molecular_subtype == "MB, Group3"~"Medulloblastoma, group 3",
                                          molecular_subtype == "MB, Group4"~ "Medulloblastoma, group 4", 
                                          TRUE ~NA_character_),
         broad_histology = "Embryonal Tumor",
         short_histology = "Medulloblastoma")
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
+                    short_histology,
+                    broad_histology,
+                    Notes),
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = ifelse(molecular_subtype == "CRANIO, ADAM",
+                                       "Adamantimomatous craniopharyngioma",
+                                       ifelse(molecular_subtype == "CRANIO, PAP","Papillary craniopharyngioma", NA)),
+         broad_histology == "Tumors of sellar region",
+         short_histology == "Craniopharyngioma")
```
```suggestion
             by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "CRANIO, ADAM" ~"Adamantimomatous craniopharyngioma",
                                          molecular_subtype == "CRANIO, PAP" ~"Papillary craniopharyngioma", 
                                          TRUE ~ NA_character_),
         broad_histology = "Tumors of sellar region",
         short_histology = "Craniopharyngioma")
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -166,30 +203,119 @@ hgg_results_df <- bind_exp_strategies(hgg_results_df) %>%
+                    short_histology,
+                    broad_histology,
+                    Notes),
+             by = "Kids_First_Biospecimen_ID") %>%
+  mutate(integrated_diagnosis = ifelse(molecular_subtype == "CNC",
+                                       "Central Neurocytoma",
+                                       ifelse(molecular_subtype == "EVN","Extraventricular Neurocytoma", NA)),
+         broad_histology == "Neuronal and mixed neuronal-glial tumor",
+         short_histology == "Neurocytoma")
```
```suggestion
            by = "Kids_First_Biospecimen_ID") %>%
  mutate(integrated_diagnosis = case_when(molecular_subtype == "CNC"~ "Central Neurocytoma",
                                          molecular_subtype == "EVN" ~"Extraventricular Neurocytoma", 
                                        TRUE ~ NA_character_),
         broad_histology = "Neuronal and mixed neuronal-glial tumor",
         short_histology = "Neurocytoma")
```

kgaonkar6 reviewed (commented):
I think we can use case_when() calls here for assigning subtypes since we have multiple options. I added some suggestions for changing ifelse calls to case_when()

jharenza commented:
> Oh yes, BS_EE73VE7V ( 1 of the sample Progressive Disease Post-Mortem for PT_KTRJ8TFY ) only had the histone variant in Vardict so is lost in consensus calling so got subtyped as H3 wildtype.
> 
> Ticket [d3b-center/bixu-tracker#463](https://github.com/d3b-center/bixu-tracker/issues/463) shows the IGV for the variant, we should probably add that to the clinical/pathology module as H3 mutatant along with other updates?

For now, since we do not have clinical sequencing to confirm this, let's leave this out of the clinical module. I expect this will be captured when #819 goes in by @migbro in a few weeks

jharenza commented:
@kgaonkar6 I updated the code based on your comments, updated Notes, and added in the 03-script the changes due to pathology feedback. I then get 1487 rows in the final file, but 1481 biospecimens. It seems like 6 more have duplicates, but when I check for duplicates of the final file using 
```
compiled_df[duplicated(compiled_df$Kids_First_Biospecimen_ID),]
```

I get the table below (which do not show me duplicate BS_IDs). (Maybe overlooking something in the code).


Kids_First_Participant_ID<chr> | sample_id<chr> | Kids_First_Biospecimen_ID<chr> | molecular_subtype<chr> | integrated_diagnosis<chr> |  
-- | -- | -- | -- | -- | --
PT_00G007DM | 7316-272 | BS_QWNBZ9RJ | MB, To be classified | NA |  
PT_00G007DM | 7316-272 | BS_K07KNTFY | MB, To be classified | NA |  
PT_9BZETM0M | 7316-158 | BS_5P1TN10Z | HGG, H3 G35 | High-grade glioma/astrocytoma, H3 G35-mutant |  
PT_9BZETM0M | 7316-158 | BS_TV5B86ZD | HGG, H3 G35 | High-grade glioma/astrocytoma, H3 G35-mutant |  
PT_9BZETM0M | 7316-158 | BS_STNH7YSX | HGG, H3 G35 | High-grade glioma/astrocytoma, H3 G35-mutant |  

kgaonkar6 commented on analyses/molecular-subtyping-pathology/03-incorporate-pathology-feedback.Rmd:
```diff
@@ -244,17 +246,235 @@ compiled_df <- compiled_df %>%
+    molecular_subtype = case_when(
+      sample_id == "7316-229" ~ "EWS",
+      TRUE ~ molecular_subtype
+    )
```
```suggestion
    ),
    Notes = case_when(sample_id == "7316-229" ~ "Updated via OpenPBTA subtyping",
      TRUE ~ Notes
    )
```

kgaonkar6 commented on analyses/molecular-subtyping-pathology/03-incorporate-pathology-feedback.Rmd:
````diff
@@ -244,17 +246,235 @@ compiled_df <- compiled_df %>%
+#PT_KTRJ8TFY (fixed in clinical feedback) and PT_6Q0NPVP3 (fixed in HGG module removing LGAT)
+```
+### `PT_00G007DM`
+
````
```suggestion
7316-272 is subtyped as ETMR because a C19MC alteration was found and pathology rerview confirms the new diagnosis

```{r}
compiled_df <- compiled_df %>%
  mutate(molecular_subtype= case_when(sample_id == "7316-272"~"ETMR, C19MC-altered",
                                      TRUE ~ molecular_subtype),
         short_histology= case_when(sample_id == "7316-272"~"ETMR",
                                      TRUE ~ short_histology),
         integrated_diagnosis = case_when(sample_id == "7316-272"~"Embryonal tumor with multilayer rosettes, C19MC-altered",
                                      TRUE ~ integrated_diagnosis)
  ) %>%
  unique()

```

Sample 7316-272 from PT_00G007DM has both MB (to be classified) and ETMR subtyping values in compiled_df. Since we know that C19MC alteration is found in this sample we would need to update the molecular_subtype/short_histology/intergated_diagnosis values to ETMR to not have duplicated rows for this sample :

Kids_First_Participant_ID<chr> | sample_id<chr> | Kids_First_Biospecimen_ID<chr> | molecular_subtype<chr> | integrated_diagnosis<chr> |  
-- | -- | -- | -- | -- | --
PT_00G007DM | 7316-272 | BS_QWNBZ9RJ | ETMR, C19MC-altered | Embryonal tumor with multilayer rosettes, C19MC-altered |  
PT_00G007DM | 7316-272 | BS_K07KNTFY | ETMR, C19MC-altered | Embryonal tumor with multilayer rosettes, C19MC-altered |  
PT_00G007DM | 7316-272 | BS_QWNBZ9RJ | MB, To be classified | NA |  
PT_00G007DM | 7316-272 | BS_K07KNTFY | MB, To be classified | NA

kgaonkar6 reviewed (commented):
Thanks @jharenza! From your comment above I also looked into 03 now and it seems the following updates will remove the duplicated rows. 

```
annotation_cols <- unlist(str_split("integrated_diagnosis,broad_histology,short_histology,molecular_subtype", ","))

get_count<-function(x){
compiled_df %>% 
    dplyr::group_by(Kids_First_Biospecimen_ID,!!as.name(x)) %>%
    tally() %>%
    filter(n>1)
}
lapply(annotation_cols,function(x) get_count(x))
```
The above check had no rows so we only have 1 value per bs_id in complied_df now which is expected.

Additionally, EPN molecular_subtypes sometimes deviated from the format of like HGG, wildtype/ HGG, K28M values do we want to update that ( in EPN updated PR) ? 
```r
# A tibble: 5 x 1
  molecular_subtype    
  <chr>                
1 EPN, To be classified
2 ST_EPN_RELA          
3 ST_EPN_YAP1          
4 EPN, H3 K28          
5 PT_EPN_A 
```

jharenza commented:
> Additionally, EPN molecular_subtypes sometimes deviated from the format of like HGG, wildtype/ HGG, K28M values do we want to update that ( in EPN updated PR) ?

Sure, sounds like we can make that update.

jharenza commented on analyses/molecular-subtyping-pathology/03-incorporate-pathology-feedback.Rmd:
```diff
@@ -244,17 +246,235 @@ compiled_df <- compiled_df %>%
+      TRUE ~ short_histology
+    ),
+    broad_histology = case_when(
+      sample_id == "7316-156" ~ "Diffuse astrocytic and oligodendroglial tumor",
```
oh gosh, good catch!

jharenza commented on analyses/molecular-subtyping-pathology/03-incorporate-pathology-feedback.Rmd:
````diff
@@ -244,17 +246,250 @@ compiled_df <- compiled_df %>%
+#PT_KTRJ8TFY (fixed in clinical feedback) and PT_6Q0NPVP3 (fixed in HGG module removing LGAT)
+```
+### `PT_00G007DM`
+7316-272 is subtyped as ETMR because a C19MC alteration was found and pathology rerview confirms the new diagnosis
````
```suggestion
7316-272 is subtyped as ETMR because a C19MC alteration was found and pathology review confirms the new diagnosis
```

jharenza replied:
fix typo

jharenza commented:
@kgaonkar6 applied changes, fixed typo and one missed chunk closure, and reran, so requested your review once more. Looks like all of the duplicates are now fixed.

kgaonkar6 reviewed (approved):
Looks good! Does all the required tasks in #854
